src/node/transaction.cpp

```diff
@@ -104,3 +107,38 @@ TransactionError BroadcastTransaction(NodeContext& node, const CTransactionRef t
 
     return TransactionError::OK;
 }
+
+CTransactionRef GetTransaction(const CBlockIndex* const block_index, const CTxMemPool* const mempool, const uint256& hash, const Consensus::Params& consensusParams, uint256& hashBlock)
+{
+    LOCK(cs_main);
```

Comment:
Maybe it's also a good opportunity to address https://github.com/bitcoin/bitcoin/pull/22383#discussion_r675069128 here?

Reply:
+1

Reply:
I'd say no. Better to keep this as move only and leave lock cleanups for a future PR. That makes this easier to review (and avoids invalidating the three ACKs on the current branch).

Reply:
The PR was planned to be one of the move-only refactoring kinds that should be trivial to review (strictly speaking, with the follow-up commit changing the RPC help text it's not a pure refactor anymore, but that I'd still consider acceptable). Changing locks is significantly harder to review, so I tend to agree with jnewbery here and leave it as it is by now.